cmd/snap/cmd_run.go

```diff
@@ -1047,7 +1122,7 @@ func (x *cmdRun) runCmdUnderStrace(origCmd []string, envForExec envForExecFunc)
 	return err
 }
 
-func (x *cmdRun) runSnapConfine(info *snap.Info, securityTag, snapApp, hook string, args []string) error {
+func (x *cmdRun) runSnapConfine(info *snap.Info, securityTag, snapApp, hook string, beforeExec func() error, args []string) error {
```

Comment:
Can you rename from `beforeExec` to `beforeMaybeExec` as the key property is that in the gdb/strace case we do _not_ exec anything in the main process.

Reply:
How bout `beforeRun`? `beforeMaybeExec` gives the impression that it might not run the callback.